components/layout/display_list_builder.rs

```diff
@@ -196,16 +198,31 @@ impl<'a> DisplayListBuildState<'a> {
         }
     }
 
+    fn update_overflow_for_stacking_context(&mut self, stacking_context: &mut StackingContext) {
+        if stacking_context.context_type != StackingContextType::Real {
+            return;
+        }
+
+        for child in self.stacking_context_children(stacking_context.id).iter_mut() {
```

Comment:
This will create a new entry in the `HashMap`, even if there wasn't any stacking context, right?

Would it be sensible to create a function that either returns an `Option`, or does something like:

```rust
pub fn iter_children_mut(&mut self, id: StackingContextId) -> &mut [StackingContext] {
    self.stacking_context_children.get_mut(id).unwrap_or(&mut [])
}
```